src/app/blogs/Blog.tsx

```diff
@@ -17,59 +25,60 @@ async function getData() {
-  const blogData = getData();
-
-  // Wait for the promises to resolve
-  const [dataT] = await Promise.all([blogData]);
+  const blogData: BlogData[] = await getData();
 
   return (
     <section className="mt-20">
       <div className="mx-auto max-w-5xl md:px-8">
-        <h1 className="mt-10 text-xl font-bold">Get to know more</h1>
+        <h1 className="mt-10 text-center text-2xl font-bold">Get to know more</h1>
         <p className="mx-auto mt-5 max-w-sm text-center md:max-w-xl lg:max-w-3xl">
           Discover the exciting world of blockchain technology as we come together to collaborate
           and expand our knowledge.
         </p>
       </div>
       <div className="mt-10 flex flex-wrap justify-center gap-16">
-        {dataT.map((curElem: any) => (
-          <div className="bg-whie flex max-w-xs flex-col rounded-lg bg-white/5 p-2 shadow-md shadow-purple-400/40 transition-all duration-200 ease-in hover:z-50 hover:shadow-lg hover:shadow-purple-400/60 sm:hover:-translate-y-1">
-            <Image
-              className="w-full rounded-md border"
-              src={curElem.cover_image}
-              width={400}
-              height={250}
-              alt=""
-            />
-            <div className="grow px-6 py-4">
-              <h3 className="mb-4 text-center text-sm font-bold uppercase leading-4 tracking-wider">
-                {curElem.title}
-              </h3>
-              <p className="text-justify text-sm text-gray-700 dark:text-gray-300">
-                {curElem.description}
-              </p>
-            </div>
-            <div className="flex px-6 py-4">
-              <Link
-                href={curElem.url}
-                target="_blank"
-                className="w-fit-content font-inter text-14.0418 group mb-0 mr-auto flex items-center gap-6 rounded border-2 border-solid border-black px-4 py-1 text-black transition delay-0 ease-in hover:border-slate-500 hover:bg-slate-500 hover:font-semibold dark:border-white/80 dark:text-white dark:hover:border-slate-500"
-              >
-                <Image
-                  src="/Arrow1.png"
-                  alt="arrow"
-                  width={15}
-                  height={15}
-                  className="my-auto flex items-center transition duration-100 ease-in group-hover:invert dark:invert"
-                />
-                <span className="transition duration-100 ease-in group-hover:text-white">
-                  Read More
-                </span>
-              </Link>
-            </div>
-          </div>
+        {blogData.map((data) => (
+          <BCard key={data.title} data={data} />
         ))}
       </div>
     </section>
   );
 }
+
+function BCard({ data }: { data: BlogData }) {
+  const defaultImg = `https://og-image-rest-generator.fly.dev/seo-banner?head=Blog&writer=Punk%20Blogger&logo=https://avatars.githubusercontent.com/u/89759498?s=200&v=4&template=facebook-minimal&title=${data.title}&author=${data.user.username}`;
```

Comment:
This is an interesting point for me, but it would be nice if this was a helper and not written in a crude way. :thinking:

Reply:
Hi @darkterminal, thanks for your feedback! I appreciate your input. Could you kindly clarify what you mean by "helper" and "crude way"? I'd love to understand your perspective better.

Reply:
Create separate reusable functions called `createDefaultImageCover` (like) in `src/lib/utils.ts` that are responsible for his behavior. You can be defined the following parameters that represent the query params in that URL.

Reply:
Okay, Sure

Reply:
Hey @darkterminal, I have made the suggested changes. Please review them.

Reply:
I am resolving the conversation, since darkterminal approved the changes.